src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -46,15 +53,31 @@ class KubeTransferStrategy implements TransferStrategy {
     @Inject
     private K8sService k8sService
 
+    @Inject
+    private CleanupStrategy cleanup
+
+    @Inject
+    @Named(TaskExecutors.IO)
+    private ExecutorService executor
+
     @Override
     BlobCacheInfo transfer(BlobCacheInfo info, List<String> command) {
         final podName = podName(info)
         final pod = k8sService.transferContainer(podName, blobConfig.s5Image, command, blobConfig)
         final terminated = k8sService.waitPod(pod, blobConfig.transferTimeout.toMillis())
         final stdout = k8sService.logsPod(podName)
-        return terminated
+        final result = terminated
                 ? info.completed(terminated.exitCode, stdout)
                 : info.failed(stdout)
+        if( cleanup.shouldCleanup(terminated.exitCode) ) {
+            CompletableFuture.supplyAsync (() ->
+                    k8sService.deletePodWhenReachStatus(
+                            podName,
+                            'Succeeded',
+                            blobConfig.podDeleteTimeout.toMillis()),
+                    executor)
+        }
```

Comment:
Let's wrap this block into a separate private method e.g. `cleanup0(exitCode)` to make more easy readble

Reply:
done